golem-api-grpc/proto/golem/apidefinition/api_definition.proto

```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
+
+package golem.apidefinition;
+
+import "golem/template/template_id.proto";
+
+message ApiDefinition {
+  oneof definition {
+    HttpApiDefinition http = 1;
+  }
+}
+
+message ApiDefinitionList {
+  oneof definitions {
+    HttpApiDefinitionList http = 1;
+  }
+}
```

Comment:
should not it be

```proto
message ApiDefinitionList {
  repeated ApiDefinition definitions = 1;
}
```
?

Reply:
Do you think that's better? that means that every element in list be different? This restricts the list to be of one type.

Reply:
I think multiple types is necessary for`rpc GetAllApiDefinitions (GetAllApiDefinitionsRequest) returns (GetAllApiDefinitionsResponse) {}`

Reply:
it is used in `GetAllApiDefinitionsResponse`, which based on my understanding, is all definitions, does not matter what type it is

Reply:
it is also used in `GetApiDefinitionVersions`, which would be all of the same type. I don't think it's worth the increased complexity, so I will go with your design